src/dune_pkg/sys_poll.ml

```diff
@@ -48,33 +48,51 @@ let run_capture_line ~path ~prog ~args =
   | Some prog -> Process.run_capture_line ~display:Quiet Strict prog args >>| norm
 ;;
 
+module Config_override_variables = struct
+  let string_option_config name =
+    let config =
+      Dune_config.Config.make ~name ~of_string:(fun s -> Ok (Some s)) ~default:None
+    in
+    fun () -> Dune_config.Config.get config
+  ;;
+
+  let os = string_option_config "os"
+  let arch = string_option_config "arch"
```

Comment:
I wonder if we should just allow overriding these variables at build rime rather than just during solving. That would replace this config and and would likely be a useful feature in general.

Reply:
This should already be the case. The overrides are applied in functions in `Sys_poll` such as `Sys_poll.os` which are used to compute variables both at solve time and build time.